src/source/Ice/IceAgent.c

```diff
@@ -1757,6 +1757,15 @@ STATUS iceAgentInitRelayCandidates(PIceAgent pIceAgent)
     return retStatus;
 }
 
+
+STATUS trunStateFailedFn(UINT64 data, PSocketConnection pSocketConnection) {
+    PIceCandidate pNewCandidate = (PIceCandidate)data;
+    if (pNewCandidate != NULL && pNewCandidate->state == ICE_CANDIDATE_STATE_NEW) {
+        pNewCandidate->state = ICE_CANDIDATE_STATE_INVALID;
+    }
+    return 0;
```

Comment:
Return STATUS_SUCCESS

Reply:
Actually, you can decorate the function like many others in the file where it checks for pNewCandidate not being null like this

CHK(pNewCandidate != NULL, STATUS_NULL_ARG);

You will need to add a CleanUp: label at the end and STATUS retStatus = STATUS_SUCCESS at the beginning